### ISSUE TITLE
On the second-order gradients of scatter_mul

### ISSUE THREAD

**Issue by speedcell4** [rusty1s/pytorch_scatter#213]
Hi. I have tried to compute the second-order gradients of `scatter_mul` by using the following scripts, 

```python
import torch
from torch_scatter import scatter_mul

if torch.cuda.is_available():
    device = torch.device('cuda:0')
else:
    device = torch.device('cpu')

x = torch.randn((5,), device=device, requires_grad=True)
index = torch.randint(0, 3, (5,), device=device)
z = scatter_mul(x, index=index, dim=0)

print(f'x => {x}')
print(f'index => {index}')
print(f'z => {z}')

grad_z, = torch.autograd.grad(
    z, x, torch.ones_like(z),
    create_graph=True,
)
print(f'grad_z => {grad_z}')

grad_grad_z, = torch.autograd.grad(
    grad_z, x, torch.ones_like(grad_z),
    create_graph=True,
)
print(f'grad_grad_z => {grad_grad_z}')

# x => tensor([ 0.5471,  0.9674, -1.5657, -0.4025,  1.1367], requires_grad=True)
# index => tensor([1, 0, 2, 0, 1])
# z => tensor([-0.3894,  0.6218, -1.5657], grad_fn=<CppNode<ScatterMul>>)
# grad_z => tensor([ 1.1367, -0.4025,  1.0000,  0.9674,  0.5471],
#        grad_fn=<MaskedFillBackward0>)
# grad_grad_z => tensor([1.0000, 1.0000, 0.0000, 1.0000, 1.0000], grad_fn=<AddBackward0>)
```

But I found the second gradients are not all zeros. Did I miss something?

**Comment by rusty1s** [comment]
I'm not entirely sure how to calculate 2nd order derivatives in PyTorch, but something like 
```python
x = torch.randn((3, 2))
z = x.prod(dim=-1)
```
will yield the same result. I guess there is something wrong with how you compute the 2nd order derivative.

**Comment by speedcell4** [comment]
Oh, I figured it out, my mistake.
Thank you for replying.

**Comment by rusty1s** [comment]
Can you show how you solved it?
